packages/did-manager/src/id-manager.ts

```diff
@@ -133,13 +135,46 @@ export class DIDManager implements IAgentPlugin {
   ): Promise<IIdentifier> {
     try {
       const providerName = provider || this.defaultProvider
+      // @ts-ignore
       const identifier = await this.store.get({ alias, provider: providerName })
       return identifier
     } catch {
       return this.didManagerCreate({ provider, alias, kms, options }, context)
     }
   }
 
+  /** {@inheritDoc @veramo/core#IDIDManager.didManagerUpdate} */
+  async didManagerUpdate(
+    { did, options }: IDIDManagerUpdateArgs,
+    context: IAgentContext<IKeyManager>,
+  ): Promise<IIdentifier> {
+    /**
+     * 1. Check if provider implements updateIdentifier
+     * 2. If not, throw
+     */
+    try {
+      /**
+       * 1. Check if the identifier is already in the store
+       * 2. If not, throw
+       * 3. If yes, resolve from ledger and construct the body
+       * 4. Get raw payload provided by `options.document`
+       * 5. Publish ledger transaction
+       * 6. Update the identifier in the store
+       * 7. Return the identifier
+       */
+      const identifier = await this.store.get({ did })
+      const identifierProvider = this.getProvider(identifier.provider)
+      const updatedIdentifier = await identifierProvider.updateIdentifier(
+        { did, options },
+        context,
+      )
+      await this.store.import(updatedIdentifier)
+      return updatedIdentifier
+    } catch {
+      throw Error('Provider does not support updateIdentifier')
+    }
```

Comment:
It's probably better to let the lower level errors bubble up here so that the developer gets the more precise failure reason, instead of catching and re-throwing as a generic error message.

If the intent is to throw when this method is not implemented, then it might be better to first check for the type before calling it:
 
```ts
      if (typeof identifierProvider?.updateIdentifier !== 'function') {
        throw new Error(`not_supported: ${identifier?.provider} provider does not implement full document updates`)
      }
      const updatedIdentifier = await identifierProvider.updateIdentifier(
      // ...
```

```suggestion
```

Reply:
Great point. Included.